include/dlaf/permutations/general.h

```diff
@@ -0,0 +1,60 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <blas.hh>
+
+#include "dlaf/common/assert.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/permutations/general/api.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf::permutations {
+
+/// Permutes the columns or rows of an input sub-matrix mat_in[i_begin:i_end][i_begin:i_end] into an
+/// output sub-matrix mat_out[i_begin:i_end][i_begin:i_end] using the index map of permutations
+/// perms[i_begin:i_end].
+///
+/// @param perms is the index map of permutations represented as a tiled column vector. Indices are in
+///        the range [0, n) where `n` is the size of the subproblem (i.e. the indices are local to the
+///        subproblem, they are not global). Only tiles whose row tile coords are in the closed range
+///        [i_begin,i_end] are accessed in read-only mode.
```

Comment:
Not sure "subproblem" in this context is the right word. If I'm not wrong, indices refers to the local part of the matrix.